server/controllers/events/events_controller.go

```diff
@@ -43,6 +43,9 @@ const bitbucketCloudRequestIDHeader = "X-Request-UUID"
 const bitbucketServerRequestIDHeader = "X-Request-ID"
 const bitbucketServerSignatureHeader = "X-Hub-Signature"
 
+// The URL used for Azure DevOps test webhooks
+const azuredevopsTestURL = "https://fabrikam.visualstudio.com/DefaultCollection/_apis/git/repositories/4bc14d40-c903-45e2-872e-0462c7748079"
```

Comment:
How come this url has to be hardcoded?

Shouldn't this value be in the test instead?

Who has access to use this url for testing purposes?

Couldn't this be mocked?

Reply:
The PR is to detect and ignore the test webhooks that come from Azure DevOps. The test requests for the 3 webhooks that Atlantis uses are in the PR description. (The tests aren't configurable, they just send the same requests every time)

The URL is a common property that exists in all three requests so we're using it to say: "If the request says it's coming from this URL: Ignore the request" 

No access is needed to this URL, we're simply using it to identify these test requests. 

Does that sound okay?

Reply:
Do you think we'll ever have to configure these urls?

Would you add some additional comments in the code to add this explaination in case people have this question in the future?